zulip/zulip/__init__.py

```diff
@@ -1072,17 +1072,20 @@ def deregister(self, queue_id: str, timeout: Optional[float] = None) -> Dict[str
             timeout=timeout,
         )
 
-    def get_profile(self, request: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
+    def get_profile(self, request: Optional[Dict[str, Any]] = None, **kwargs: Any) -> Dict[str, Any]:
         '''
             Example usage:
 
             >>> client.get_profile()
             {u'user_id': 5, u'full_name': u'Iago', u'short_name': u'iago', ...}
         '''
+        if request is not None:
+            logging.warning('Passing parameters as a dictionary will be deprecated in future versions.')
```

Comment:
Due to the duplication I would extract this as a very simple function with a descriptive name., or at minimum the string.

Regarding the message, aren't we referring to particularly *request* parameters in a dictionary being deprecated?

Reply:
“Will be removed”, not “will be deprecated”. It’s deprecated now (which [means](https://en.wikipedia.org/wiki/Deprecation#Software) that it still exists for compatibility, but users should stop using it); it will be removed in the future.

Reply:
Oh Sorry , My bad .